base/stacktraces.jl

```diff
@@ -52,8 +52,9 @@ struct StackFrame # this type should be kept platform-agnostic so that profiles
     file::Symbol
     "the line number in the file containing the execution context"
     line::Int
-    "the MethodInstance or CodeInfo containing the execution context (if it could be found)"
-    linfo::Union{MethodInstance, CodeInfo, Nothing}
+    "the MethodInstance or CodeInfo containing the execution context (if it could be found), \
+     or Module (for macro expansions)"
+    linfo::Union{MethodInstance, Method, Module, CodeInfo, Nothing}
```

Comment:
Is it really necessary to allow `linfo::Module`? From what I can tell, this module information is never used or explicitly tested, so there's no need to store it here. I'm concerned that this heavy-union field definition complicates the code unnecessarily.

Reply:
It's possible it's out of date, but at least at one point it was needed to look up the module for `macro expansion` frame.

Reply:
We should have implemented test cases for these new features. Do you have any use case for that information? Otherwise we should clean that up.

Reply:
Ok. But I just described the use case. Is this causing a problem somewhere right now? Otherwise something to look at when the underlying problem #50204 is resolved someway and the full functionality can be restored here.